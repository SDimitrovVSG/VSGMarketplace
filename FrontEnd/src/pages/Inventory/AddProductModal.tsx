import {
    TextField,
    FormControl,
    InputLabel,
    Select,
    MenuItem,
    CircularProgress,
    FormHelperText,
} from '@mui/material';
import { ICategory, IFormInputs, ILocation, IProduct } from '../../types/types.ts';
import { useState, Dispatch, SetStateAction } from 'react';
import { useCreateProductMutation } from '../../services/productsService.ts';
import { useCreateImageMutation } from '../../services/imageService.ts';
import { useGetCategoriesQuery } from '../../services/categoriesService.ts';
import { useGetLocationQuery } from '../../services/locationsService.ts';
import { imagePlaceholder } from '../../utils/imagePlaceholder.ts';
import { uploadImage } from '../../utils/uploadImage.ts';
import { useForm } from 'react-hook-form';
import { toast } from 'react-toastify';
import Modal from '../../components/Modal.tsx';

type AddModalProps = {
    products: IProduct[] | undefined;
    setProducts: Dispatch<SetStateAction<IProduct[]>>;
    showAddModal: boolean;
    setShowAddModal: Dispatch<SetStateAction<boolean>>;
};

const AddProductModal = ({
    //products,
    setProducts,
    showAddModal,
    setShowAddModal,
}: AddModalProps) => {
    const [createProduct] = useCreateProductMutation();
    const [createImage] = useCreateImageMutation();
    const { data: categories } = useGetCategoriesQuery();
    const { data: locations } = useGetLocationQuery();
    const [imageUrl, setImageUrl] = useState(imagePlaceholder);
    const [selectOption, setSelectOption] = useState('');
    const [locationOption, setLocationOption] = useState('');
    const {
        register,
        handleSubmit,
        formState: { errors, isSubmitting },
        getValues,
        reset,
        watch
    } = useForm<IFormInputs>({
        defaultValues: {
            code: '',
            fullName: '',
            description: '',
            categoryId: null,
            locationId: null,
            quantityForSale: null,
            price: null,
            quantity: null,
            image: null,
        },
        mode: 'onChange',
    });

    const onSubmit = async (data: IFormInputs): Promise<void> => {
        const image: { name: string } | File = data.image ? data.image[0] : { name: '' };
        data.quantityForSale = data.quantityForSale || null;
        data.price = data.price || null;

        const response = await createProduct(data) as { data: IProduct };
        if ('error' in response) {
            setShowAddModal(false);
            return;
        }

        const responseData = response.data;
        const selectedCategory = categories?.filter(c => data.categoryId === c.id)[0] as ICategory;
        const selectedCity = locations?.filter(l => data.locationId === l.id)[0] as ILocation;
        const id = responseData.id;

        if (image.name) {
            const imageForm = new FormData();
            imageForm.set('picture', image as File);
            const imgRes = await createImage({ id, imageForm }) as { data: string };
            const newImgUrl = imgRes.data;
            const newProduct = { ...data, id: id, type: selectedCategory.type, city: selectedCity.city, imageUrl: newImgUrl } as IProduct;
            setProducts((oldProducts) => [...oldProducts, newProduct]);
        } else {
            const newProduct = { ...data, id: id, type: selectedCategory.type, city: selectedCity.city } as IProduct;
            setProducts((oldProducts) => [...oldProducts, newProduct]);
        }

        setSelectOption('');
        setLocationOption('');
        toast.success('Created successfully!');
        setShowAddModal(false);
        setImageUrl(imagePlaceholder);
        reset();
    };

    return (
        <Modal showModal={showAddModal} setShowModal={setShowAddModal}>
            <form className='modalContent' onSubmit={handleSubmit(onSubmit)}>
                <div className='row'>
                    <div className='leftModal'>
                        <h2>Add New Item</h2>
                        <TextField
                            className='formInput'
                            type='text'
                            label='Code *'
                            variant='standard'
                            error={Boolean(errors.code)}
                            helperText={errors.code?.message}
                            {...register('code', {
                                required: 'Code field is required',
                                // validate: value => products ? products.some(p => p.code !== value) : false || 'Code already exists'
                            })}
                        />
                        <TextField
                            className='formInput'
                            type='text'
                            label='Name *'
                            variant='standard'
                            error={Boolean(errors.fullName)}
                            helperText={errors.fullName?.message}
                            {...register('fullName', { required: 'Name field is required' })}
                        />
                        <TextField
                            className='description'
                            label='Description'
                            multiline
                            rows={4}
                            variant='standard'
                            {...register('description')}
                        />
                        <FormControl variant='standard' className='formInput'>
                            <InputLabel focused={false}>Category *</InputLabel>
                            <Select
                                value={selectOption}
                                error={(Boolean(errors.categoryId) && Boolean(errors.categoryId?.message))}
                                {...register('categoryId', {
                                    onChange: (e) => setSelectOption(e.target.value),
                                    required: 'Category field is required',
                                })}
                            >
                                {categories?.map((c: ICategory) => (
                                    <MenuItem value={c.id} key={c.id}>
                                        {c.type}
                                    </MenuItem>
                                ))}
                            </Select>
                            <FormHelperText error>
                                {watch('categoryId') === null && errors.categoryId?.message ? errors.categoryId?.message : ''}
                            </FormHelperText>
                        </FormControl>
                        <FormControl variant='standard' className='formInput'>
                            <InputLabel focused={false}>Location *</InputLabel>
                            <Select
                                value={locationOption}
                                error={Boolean(errors.locationId)}
                                {...register('locationId', {
                                    onChange: (e) => setLocationOption(e.target.value),
                                    required: 'Location field is required',
                                })}
                            >
                                {locations?.map((l: ILocation) => (
                                    <MenuItem value={l.id} key={l.id}>
                                        {l.city}
                                    </MenuItem>
                                ))}
                            </Select>
                            <FormHelperText error>
                                {watch('locationId') === null && errors.locationId?.message ? errors.locationId?.message : ''}
                            </FormHelperText>
                        </FormControl>
                        <TextField
                            className='formInput'
                            type='number'
                            label='Qty For Sale'
                            variant='standard'
                            error={Boolean(errors.quantityForSale)}
                            helperText={errors.quantityForSale?.message}
                            {...register('quantityForSale', {
                                min: {
                                    value: 0,
                                    message: 'You cannot input less than 0'
                                },
                            })}
                        />
                        <TextField
                            className='formInput'
                            type='number'
                            label='Sale Price'
                            variant='standard'
                            error={Boolean(errors.price)}
                            helperText={errors.price?.message}
                            {...register('price', {
                                min: {
                                    value: 0,
                                    message: 'You cannot input less than 0'
                                }
                            })}
                        />
                        <TextField
                            className='formInput'
                            type='number'
                            label='Qty *'
                            variant='standard'
                            error={Boolean(errors.quantity)}
                            helperText={errors.quantity?.message}
                            {...register('quantity', {
                                required: 'Quantity field is required',
                                min: {
                                    value: 0,
                                    message: 'Quantity must be 0 or higher'
                                },
                                validate: value => value as number >= Number(getValues('quantityForSale')) || 'Qty For Sale cannot be higher than Qty'
                            })}
                        />
                    </div>
                    <div className='rightModal'>
                        <img className='currentImg' src={imageUrl} />
                        <input
                            id='uploadInput'
                            className='inputImage'
                            accept='image/*'
                            type='file'
                            {...register('image', {
                                onChange: (e) => setImageUrl(uploadImage(e))
                            })}
                        />
                        <div className='uploadDelete'>
                            <label htmlFor='uploadInput' className='uploadImg'>
                                Upload
                            </label>
                            <button type='button' className='deleteImg' onClick={() => setImageUrl(imagePlaceholder)}>Remove</button>
                        </div>
                    </div>
                </div>
                {isSubmitting
                    ? <CircularProgress className='circular' />
                    : <button type='submit'>Add</button>
                }
            </form>
        </Modal>
    );
};

export default AddProductModal;
