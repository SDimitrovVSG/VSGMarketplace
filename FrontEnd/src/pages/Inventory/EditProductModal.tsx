import { useDeleteImageMutation, useEditImageMutation } from '../../services/imageService.ts';
import { useState, Dispatch, SetStateAction, } from 'react';
import { IFormInputs, ICategory, IProduct, ILocation } from '../../types/types.ts';
import { useEditProductMutation } from '../../services/productsService.ts';
import { useGetCategoriesQuery } from '../../services/categoriesService.ts';
import { useGetLocationQuery } from '../../services/locationsService.ts';
import { imagePlaceholder } from '../../utils/imagePlaceholder.ts';
import { uploadImage } from '../../utils/uploadImage.ts';
import { useForm } from 'react-hook-form';
import { toast } from 'react-toastify';
import Modal from '../../components/Modal.tsx';
import {
    TextField,
    FormControl,
    InputLabel,
    Select,
    MenuItem,
    FormHelperText,
} from '@mui/material';

type EditModalProps = {
    setProducts: Dispatch<SetStateAction<IProduct[]>>;
    product: IProduct;
    showEditModal: boolean;
    setShowEditModal: Dispatch<SetStateAction<boolean>>;
};

const EditProductModal = ({ setProducts, product, showEditModal, setShowEditModal }: EditModalProps) => {
    const [locationOption, setLocationOption] = useState(product.locationId.toString());
    const [selectOption, setSelectOption] = useState(product.categoryId.toString());
    const [imageUrl, setImageUrl] = useState(product.imageUrl || imagePlaceholder);
    const [editProduct, { isLoading: fetchingProduct }] = useEditProductMutation();
    const [editImage, { isLoading: fetchingImage }] = useEditImageMutation();
    const [deleteImage] = useDeleteImageMutation();
    const { data: categories } = useGetCategoriesQuery();
    const { data: locations } = useGetLocationQuery();
    const {
        register,
        handleSubmit,
        formState: { errors },
        getValues,
    } = useForm<IFormInputs>({
        defaultValues: {
            code: product.code,
            fullName: product.fullName,
            description: product.description,
            categoryId: product.categoryId,
            locationId: product.locationId,
            quantityForSale: product.quantityForSale || null,
            price: product.price || null,
            quantity: product.quantity,
            image: null
        },
        mode: 'all',
    });

    const onSubmit = async (data: IFormInputs) => {
        const image: { name: string } | File = data.image ? data.image[0] : { name: '' };
        data.quantityForSale = data.quantityForSale || null;
        data.price = data.price || null;

        const currentImg = document.querySelector('.currentImg') as HTMLImageElement;
        const selectedCategory = categories?.filter(c => data.categoryId === c.id)[0] as ICategory;
        const selectedCity = locations?.filter(l => data.locationId === l.id)[0] as ILocation;
        const id = product.id;

        if (image.name) {
            const imageForm = new FormData();
            imageForm.append('newPicture', image as File);
            const imgRes = await editImage({ id, imageForm }) as { data: string };
            const newImgUrl = imgRes.data;
            setImageUrl(newImgUrl);
            const editedProduct = { ...data, id: id, type: selectedCategory.type, city: selectedCity.city, imageUrl: newImgUrl } as IProduct;
            setProducts(oldProducts => oldProducts.map(p => p.id === editedProduct.id ? editedProduct : p));
        } else if (currentImg.src !== product.imageUrl && product.imageUrl !== null) {
            await deleteImage(id);
        }

        const response = await editProduct({ id, data }) as { data: IProduct };

        if ('error' in response) {
            setShowEditModal(false);
            return
        } else {
            toast.success('Modified successfully!');
        }

        const editedProduct = { ...data, id: id, type: selectedCategory.type, city: selectedCity.city } as IProduct;
        setProducts(oldProducts => oldProducts.map(p => p.id === editedProduct.id ? editedProduct : p));
        setShowEditModal(false);
    };

    return (
        <Modal showModal={showEditModal} setShowModal={setShowEditModal} >
            <form className='editForm' onSubmit={handleSubmit(onSubmit)}>
                <div className='row'>
                    <div className='leftModal'>
                        <h2>Modify Item</h2>
                        <TextField
                            className='formInput'
                            type='text'
                            label='Code *'
                            variant='standard'
                            error={Boolean(errors.code)}
                            helperText={errors.code?.message}
                            {...register('code', { required: 'Code field is required' })}
                        />
                        <TextField
                            className='formInput'
                            type='text'
                            label='Name *'
                            variant='standard'
                            error={Boolean(errors.fullName)}
                            helperText={errors.fullName?.message}
                            {...register('fullName', { required: 'Name field is required' })}
                        />
                        <TextField
                            className='description'
                            label='Description'
                            multiline
                            rows={4}
                            variant='standard'
                            {...register('description')}
                        />
                        <FormControl variant='standard' className='formInput'>
                            <InputLabel focused={false}>Category *</InputLabel>
                            <Select
                                value={selectOption}
                                error={Boolean(errors.categoryId)}
                                {...register('categoryId', {
                                    onChange: (e) => setSelectOption(e.target.value),
                                    required: 'Category field is required',
                                })}
                            >
                                <MenuItem value={selectOption} key={selectOption}>
                                    {product.type}
                                </MenuItem>
                                {categories?.map((c: ICategory) => (
                                    c.type !== product.type &&
                                    <MenuItem value={c.id} key={c.id}>
                                        {c.type}
                                    </MenuItem>
                                ))}
                            </Select>
                        </FormControl>
                        <FormControl variant='standard' className='formInput'>
                            <InputLabel focused={false}>Location *</InputLabel>
                            <Select
                                value={locationOption}
                                error={Boolean(errors.locationId)}
                                {...register('locationId', {
                                    onChange: (e) => setLocationOption(e.target.value),
                                    required: 'Location field is required',
                                })}
                            >
                                <MenuItem value={locationOption} key={locationOption}>
                                    {product.city}
                                </MenuItem>
                                {locations?.map((l: ILocation) => (
                                    l.city !== product.city &&
                                    <MenuItem value={l.id} key={l.id}>
                                        {l.city}
                                    </MenuItem>
                                ))}
                            </Select>
                            <FormHelperText error>
                                {errors.locationId?.message}
                            </FormHelperText>
                        </FormControl>
                        <TextField
                            className='formInput'
                            type='number'
                            label='Qty For Sale'
                            variant='standard'
                            error={Boolean(errors.quantityForSale)}
                            helperText={errors.quantityForSale?.message}
                            {...register('quantityForSale', {
                                min: {
                                    value: 0,
                                    message: 'You cannot input less than 0'
                                },
                            })}
                        />
                        <TextField
                            className='formInput'
                            type='number'
                            label='Sale Price'
                            variant='standard'
                            {...register('price', {
                                min: {
                                    value: 0,
                                    message: 'You cannot input less than 0'
                                }
                            })}
                        />
                        <TextField
                            className='formInput'
                            type='number'
                            label='Qty *'
                            variant='standard'
                            error={Boolean(errors.quantity)}
                            helperText={errors.quantity?.message}
                            {...register('quantity', {
                                required: 'Quantity field is required',
                                min: {
                                    value: 0,
                                    message: 'Quantity must be 0 or higher'
                                },
                                validate: value => value as number >= Number(getValues('quantityForSale')) || 'Qty cannot be less than Qty For Sale'
                            })}
                        />
                    </div>
                    <div className='rightModal'>
                        <img className='currentImg' src={imageUrl} />
                        <input
                            id='editUploadInput'
                            className='inputImage'
                            accept='image/*'
                            type='file'
                            {...register('image', {
                                onChange: (e) => setImageUrl(uploadImage(e))
                            })}
                        />
                        <div className='uploadDelete'>
                            <label htmlFor='editUploadInput' className='uploadImg'>
                                Upload
                            </label>
                            <button type='button' className='deleteImg' onClick={() => setImageUrl(imagePlaceholder)}>Remove</button>
                        </div>
                    </div>
                </div>
                <button type='submit' disabled={fetchingProduct || fetchingImage}>{(fetchingProduct || fetchingImage) ? 'Submitting...' : 'Modify'}</button>
            </form>
        </Modal>
    );
};


export default EditProductModal;
