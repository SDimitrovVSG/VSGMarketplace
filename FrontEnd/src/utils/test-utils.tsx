import { ReactElement } from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';

import store from '../redux/store';
import { BrowserRouter } from 'react-router-dom';

type ProvidersProps = {
  children: ReactElement;
};

const AllTheProviders = ({ children }: ProvidersProps) => {
  return (
    <Provider store={store}>
      <BrowserRouter>
        {children}
      </BrowserRouter>
    </Provider>
  )
}

const customRender = (ui: ReactElement, options?: object) => (
  render(ui, { wrapper: AllTheProviders, ...options })
);

export * from '@testing-library/react';

// override render method
export { customRender as render }