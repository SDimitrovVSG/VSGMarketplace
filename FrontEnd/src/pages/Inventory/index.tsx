import { useGetInventoryQuery } from "../../services/productsService";
import { useGetLocationQuery } from "../../services/locationsService.ts";
import { ILocation, IProduct } from "../../types/types.ts";
import { useState, useEffect } from "react";
import AddProductModal from "./AddProductModal.tsx";
import LocationOnIcon from '@mui/icons-material/LocationOn';
import InventoryTable from "./InventoryTable.tsx";
import SearchIcon from "@mui/icons-material/Search";
import AddIcon from "@mui/icons-material/Add";
import {
    Box,
    TextField,
    FormControl,
    InputLabel,
    Select,
    MenuItem,
} from "@mui/material";

const Inventory = () => {
    const { data: locations } = useGetLocationQuery();
    const { data, isLoading } = useGetInventoryQuery();
    const [locationOption, setLocationOption] = useState(0);
    const [products, setProducts] = useState<IProduct[]>([]);
    const [showAddModal, setShowAddModal] = useState(false);
    const [searchString, setSearchString] = useState("");

    useEffect(() => {
        if (data) {
            if (!locationOption) {
                setProducts(data);
            } else {
                const filteredByLocation = data.filter(p => p.locationId === Number(locationOption));
                setProducts(filteredByLocation);
            }
        }
    }, [data, locationOption]);

    const filteredProducts = products.filter((p) =>
        p.fullName.toLowerCase().includes(searchString.toLowerCase())
    );

    return (
        <>
            <AddProductModal setProducts={setProducts} showAddModal={showAddModal} setShowAddModal={setShowAddModal} />
            <main id="inventoryMain">
                <div id="searchSpace">
                    <Box className='searchProduct'>
                        <SearchIcon />
                        <TextField label="Search..." variant="standard"
                            onInput={(e) => setSearchString((e.target as HTMLInputElement).value)}
                        />
                    </Box>
                    <Box className='location' >
                        <LocationOnIcon />
                        <FormControl variant='standard'>
                            <InputLabel>Location</InputLabel>
                            <Select
                                value={locationOption}
                                onChange={(e) => setLocationOption(Number(e.target.value))}
                            >
                                <MenuItem value='0' key='0'>All Locations</MenuItem>
                                {locations?.map((l: ILocation) => (
                                    <MenuItem value={l.id} key={l.id}>
                                        {l.city}
                                    </MenuItem>
                                ))}
                            </Select>
                        </FormControl>
                    </Box>
                    <button id="addButton" onClick={() => setShowAddModal(true)}>
                        <div id='addButtonSpace'>
                            <AddIcon />
                            <span>Add new</span>
                        </div>
                    </button>
                </div>
                {products &&
                    <InventoryTable
                        isLoading={isLoading}
                        setProducts={setProducts}
                        filteredProducts={filteredProducts}
                    />
                }
            </main>
        </>
    );
};

export default Inventory;
