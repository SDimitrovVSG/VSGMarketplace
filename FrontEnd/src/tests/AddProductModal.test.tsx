import Inventory from '../pages/Inventory';
import userEvent from '@testing-library/user-event';
import mockServer from 'mock-server';
import { render, screen } from 'test-utils';

describe('AddProductModal', () => {
  beforeAll(() => mockServer.listen());

  afterEach(() => mockServer.resetHandlers());

  afterAll(() => mockServer.close());
  it('should render AddProductModal component', async () => {
    render(<Inventory />);
    const user = userEvent.setup();
    const addBtn = screen.getByText('Add new');
    await user.click(addBtn);

    const addModal = await screen.findByRole('dialog');
    expect(addModal).toBeInTheDocument();
    expect(addModal).toMatchSnapshot();
  });
});