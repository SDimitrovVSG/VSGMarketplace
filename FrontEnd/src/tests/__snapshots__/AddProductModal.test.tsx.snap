// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddProductModal should render AddProductModal component 1`] = `
<div
  aria-labelledby=":r0:"
  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-ccpzmr-MuiPaper-root-MuiDialog-paper"
  role="dialog"
>
  <button
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium closeIcon css-78trlr-MuiButtonBase-root-MuiIconButton-root"
    tabindex="0"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="CloseIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
      />
    </svg>
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
  <form
    class="modalContent"
  >
    <div
      class="row"
    >
      <div
        class="leftModal"
      >
        <h2>
          Add New Item
        </h2>
        <div
          class="MuiFormControl-root MuiTextField-root formInput css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for=":rg:"
            id=":rg:-label"
          >
            Code *
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-v4u5dn-MuiInputBase-root-MuiInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
              id=":rg:"
              name="code"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root formInput css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for=":rh:"
            id=":rh:-label"
          >
            Name *
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-v4u5dn-MuiInputBase-root-MuiInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
              id=":rh:"
              name="fullName"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root description css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for=":ri:"
            id=":ri:-label"
          >
            Description
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-multiline css-1rb63tl-MuiInputBase-root-MuiInput-root"
          >
            <textarea
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline css-66dh3a-MuiInputBase-input-MuiInput-input"
              id=":ri:"
              name="description"
              rows="4"
              style="height: 0px; overflow: hidden;"
            />
            <textarea
              aria-hidden="true"
              class="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline css-66dh3a-MuiInputBase-input-MuiInput-input"
              readonly=""
              style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding: 0px; width: 100%;"
              tabindex="-1"
            />
          </div>
        </div>
        <div
          class="MuiFormControl-root formInput css-1nrlq1o-MuiFormControl-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
          >
            Category *
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl  css-m5hdmq-MuiInputBase-root-MuiInput-root-MuiSelect-root"
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              class="MuiSelect-select MuiSelect-standard MuiInputBase-input MuiInput-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
              role="button"
              tabindex="0"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </div>
            <input
              aria-hidden="true"
              aria-invalid="false"
              class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              tabindex="-1"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
          <p
            class="MuiFormHelperText-root Mui-error MuiFormHelperText-sizeMedium css-1d1r5q-MuiFormHelperText-root"
          />
        </div>
        <div
          class="MuiFormControl-root formInput css-1nrlq1o-MuiFormControl-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
          >
            Location *
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl  css-m5hdmq-MuiInputBase-root-MuiInput-root-MuiSelect-root"
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              class="MuiSelect-select MuiSelect-standard MuiInputBase-input MuiInput-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
              role="button"
              tabindex="0"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </div>
            <input
              aria-hidden="true"
              aria-invalid="false"
              class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              tabindex="-1"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
          <p
            class="MuiFormHelperText-root Mui-error MuiFormHelperText-sizeMedium css-1d1r5q-MuiFormHelperText-root"
          />
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root formInput css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for=":rj:"
            id=":rj:-label"
          >
            Qty For Sale
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-v4u5dn-MuiInputBase-root-MuiInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
              id=":rj:"
              name="quantityForSale"
              type="number"
              value=""
            />
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root formInput css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for=":rk:"
            id=":rk:-label"
          >
            Qty For Lend
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-v4u5dn-MuiInputBase-root-MuiInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
              id=":rk:"
              name="quantityForLend"
              type="number"
              value=""
            />
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root formInput css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for=":rl:"
            id=":rl:-label"
          >
            Sale Price
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-v4u5dn-MuiInputBase-root-MuiInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
              id=":rl:"
              name="price"
              type="number"
              value=""
            />
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root formInput css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for=":rm:"
            id=":rm:-label"
          >
            Qty *
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-v4u5dn-MuiInputBase-root-MuiInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
              id=":rm:"
              name="quantity"
              type="number"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="rightModal"
      >
        <img
          class="currentImg"
          src="/images/inventory/no-image-placeholder.png"
        />
        <input
          accept="image/*"
          class="inputImage"
          id="uploadInput"
          name="image"
          type="file"
        />
        <div
          class="uploadDelete"
        >
          <label
            class="uploadImg"
            for="uploadInput"
          >
            Upload
          </label>
          <button
            class="deleteImg"
            type="button"
          >
            Remove
          </button>
        </div>
      </div>
    </div>
    <button
      type="submit"
    >
      Add
    </button>
  </form>
</div>
`;
